packages/patternfly-4/react-core/src/components/Pagination/PaginationOptionsMenu.tsx

```diff
@@ -0,0 +1,151 @@
+import paginationStyles from '@patternfly/react-styles/css/components/Pagination/pagination';
+import { css } from '@patternfly/react-styles';
+import { Dropdown, DropdownItem, DropdownDirection } from '../Dropdown';
+import { CheckIcon } from '@patternfly/react-icons';
+import { OptionsToggle } from './OptionsToggle';
+import { ToggleTemplateProps } from './ToggleTemplate';
+import { PerPageOptions } from './Pagination';
+
+export interface PaginationOptionsMenuProps extends React.HTMLProps<HTMLDivElement> {
+  /** Custom class name added to the Pagination Options Menu */
+  className?: string;
+  /** Id added to the title of the Pagination Options Menu */
+  widgetId?: string;
+  /** Menu will open up or open down from the Options menu toggle */
+  dropDirection?: 'up' | 'down';
+  /** Array of titles and values which will be the options on the Options Menu dropdown */
+  perPageOptions?: PerPageOptions[];
+  /** The Title of the Pagination Options Menu */
+  itemsPerPageTitle?: string;
+  /** The suffix to be displayed after each option on the Options Menu dropdown */
+  perPageSuffix?: string;
+  /** The type or title of the items being paginated */
+  itemsTitle?: string;
+  /** The text to be displayed on the Options Toggle */
+  optionsToggle?: string;
+  /** The total number of items being paginated */
+  itemCount?: number;
+  /** The first index of the items being paginated */
+  firstIndex?: number;
+  /** The last index of the items being paginated */
+  lastIndex?: number;
+  /** The number of items to be displayed per page */
+  perPage?: number;
+  /** This will be shown in pagination toggle span. You can use firstIndex, lastIndex, itemCount, itemsTitle props. */
+  toggleTemplate?: ((props: ToggleTemplateProps) => React.ReactElement) | string;
+  /** Function called when user selects number of items per page. */
+  onPerPageSelect?: (event: React.MouseEvent<HTMLAnchorElement>|React.KeyboardEvent, perPage: number) => void;
+}
+
+interface PaginationOptionsMenuState {
+  isOpen: boolean;
+}
+
+export class PaginationOptionsMenu extends React.Component<PaginationOptionsMenuProps, PaginationOptionsMenuState> {
+
+  static defaultProps = {
+   className: '',
+   widgetId: '',
+   dropDirection: DropdownDirection.down,
+   perPageOptions: [] as PerPageOptions[],
+   itemsPerPageTitle: 'Items per page',
+   perPageSuffix: 'per page',
+   optionsToggle: 'Select',
+   perPage: 0,
+   firstIndex: 0,
+   lastIndex: 0,
+   itemCount: 0,
+   itemsTitle: 'items',
+   toggleTemplate: ({firstIndex, lastIndex, itemCount, itemsTitle}:ToggleTemplateProps) => (
+     <React.Fragment>
+       <strong>
+         {firstIndex} - {lastIndex}
+       </strong>{' '}
+       of<strong>{itemCount}</strong> {itemsTitle}
+     </React.Fragment>
+   ),
+   onPerPageSelect: () => null as any
+ };
+
+  constructor(props: PaginationOptionsMenuProps){
+    super(props);
+    this.state = {
```

Comment:
You can move `state` out of the constructor and into a class member to also get rid of `PaginationOptionsMenuState`.

Reply:
I changed my mind, this is fine.